docs/scripts_md/MRI.md

```diff
@@ -0,0 +1,366 @@
+name and (for calibration data) scannerID.
+
+INPUT: patient name, scanner ID and database handle reference
+
+RETURNS: a reference to a hash containing elements including 'CandID',
+'visitLabel' and 'visitNo', or, in the case of failure, `undef`
+
+### subjectIDIsValid($CandID, $PSCID, $dbhr)
+
+Verifies that the subject IDs match.
+
+INPUT: candidate's CandID, candidate's PSCID and the database handle reference
+
+RETURNS: 1 if the ID pair matches, 0 otherwise
+
+### subjectIDExists($CandID, $dbhr)
+
+Verifies that the subject ID (CandID) exists.
+
+INPUT: candidate's CandID and the database handle reference
+
+RETURNS: 1 if the ID exists, 0 otherwise
+
+### getScannerCandID($scannerID, $dbhr)
+
+Retrieves the candidate (CandID) for the given scanner.
+
+INPUT: the scanner ID and the database handle reference
+
+RETURNS: the CandID or (if none exists) undef
+
+### getSessionID($subjectIDref, $studyDate, $dbhr, $objective, ...)
+
+Gets (or creates) the session ID, given CandID and visitLabel (contained
+inside the hashref `$subjectIDref`).  Unless `$noStagingCheck` is true, it
+also determines whether staging is required using the `$studyDate`
+(formatted YYYYMMDD) to determine whether staging is required based on a
+simple algorithm:
+
+> \- If there exists a session with the same visit label, then that is
+>    the session ID to use.  If any dates (either existing MRI data or
+>    simply a date of visit) exist associated with that session, then
+>    if they are outside of some (arbitrary) time window, staging is
+>    required.  If no dates exist, no staging is required.
+>
+> \- If no sessions exist, then if there is any other date associated
+>    with another session of the same subject within a time window,
+>    staging is required.
+>
+> \- Otherwise, staging is not required.
+
+INPUT: hash reference of subject IDs, study date, database handle reference,
+the objective of the study and a no staging check flag.
+
+RETURNS: a list of two items, (sessionID, requiresStaging)
+
+### checkMRIStudyDates($studyDateJD, $dbhr, @fileIDs)
+
+This method tries to figure out if there may have been labelling problems which
+would put the files in a staging area that does not actually exist.
+
+INPUT: study date, database handle reference and array of fileIDs to check the
+study date
+
+RETURNS: 1 if the file requires staging, 0 otherwise
+
+### getObjective($subjectIDsref, $dbhr)
+
+Attempts to determine the SubprojectID  of a timepoint given the subjectIDs
+hashref `$subjectIDsref` and a database handle reference `$dbhr`
+
+INPUT: subjectIDs hashref and database handle reference
+
+RETURNS: the determined objective, or 0
+
+### identify\_scan\_db($center\_name, $objective, $fileref, $dbhr)
+
+Determines the type of the scan described by minc headers based on mri\_protocol
+table in the database.
+
+INPUT: center's name, objective of the study, file hashref, database handle
+reference
+
+RETURNS: textual name of scan type
```

Comment:
maybe here be more specific and say Scan Type from mri_scan_type table? because this is one of the most important routines (in my opinion anyway)...